node/node.go

```diff
@@ -536,3 +547,27 @@ func (n *Node) DialSeeds(seeds []string) error {
 }
 
 //------------------------------------------------------------------------------
+
+var (
+	genesisDocKey = []byte("genesisDoc")
+)
+
+func loadGenesisDoc(db dbm.DB) (*types.GenesisDoc, error) {
+	bytes := db.Get(genesisDocKey)
+	if len(bytes) == 0 {
+		return nil, errors.New("Genesis doc not found")
+	} else {
+		var genDoc *types.GenesisDoc
+		err := json.Unmarshal(bytes, &genDoc)
```

Comment:
maybe we panic here actually?

Reply:
you're right

Reply:
also the double pointer here can be confusing. Isn't it better to:

```
genDoc := new(types.GenesisDoc)
Unmarshal(bytes, genDoc)
```

?

Reply:
mm.. using `*smth` seems more standard. I am not used to using `new`